ramanujan/utils/utils.py

```diff
@@ -0,0 +1,146 @@
+    """
+    create a series of type n(n(...(a[0]*n + a[1]) + a[2]) + ...) + a[k]
+    :param poly_a: a[k] coefficients
+    :param n: length of series
+    :return: a list of numbers in series
+    """
+    for i in range(starting_n, max_runs + starting_n):
+        tmp = 0
+        for c in poly_a:
+            tmp *= i
+            tmp += c
+        yield tmp
+
+
+def create_mpf_const_generator(sym_constants):
+    """
+    Returns a generator that creates an mpf objects from sympy constants
+    This allows us to get an object that matches the scope's mpf's workdps
+    """
+    constants_generator = []
+    for i in range(len(sym_constants)):
+        try:
+            constants_generator.append(lambdify((), sym_constants[i], modules="mpmath"))
+        except AttributeError:  # Hackish constant
+            constants_generator.append(sym_constants[i].mpf_val)
+    return constants_generator
+
+
+def plot_gcf_convergens(an_poly_coef, bn_poly_coef, max_iters, divide_interval=101, label=None):
+    computed_values = []
+    label = f'an {an_poly_coef} bn {bn_poly_coef}' if not label else label
+
+    # a_n iterator is always one ahead, while a[0] goes to p. so this is 
+    # an ugly hack
+
+    poly_a_deg, _ = get_poly_deg_and_leading_coef(an_poly_coef)
+    poly_b_deg, _ = get_poly_deg_and_leading_coef(bn_poly_coef)
+
+    print(f'deg a - {poly_a_deg} deg b - {poly_b_deg}')
+    if poly_a_deg * 2 == poly_b_deg:
+        print('\texpo')
+        print(f'\t4b_d = {bn_poly_coef[0] * 4} | -a^2 = {-1 * (an_poly_coef[0]**2)}')
+        if bn_poly_coef[0] * 4 > -1 * (an_poly_coef[0]**2):
+            print('\t\tcond passed')
+        elif bn_poly_coef[0] * 4 == -1 * (an_poly_coef[0]**2):
+            print('\t\tequality')
+        else:
+            print('\t\tcond failed')
+    elif poly_a_deg * 2 > poly_b_deg:
+        print('\tsuper expo')
+    else:
+        print('\tsub expo')
+    an_items_iterator = iter_series_items_from_compact_poly(an_poly_coef, max_iters)
+    bn_items_iterator = iter_series_items_from_compact_poly(bn_poly_coef, max_iters, 1)
+    
+    prev_q = 0
+    q = 1
+    prev_p = 1
+    p = an_items_iterator.__next__() # will place a[0] to p
+
+    for i, (a_i_1, b_i) in enumerate(zip(an_items_iterator, bn_items_iterator)):
+        # a_i_1 is the (i+1)'th item of an, and b_i the the i'th item of bn
+        tmp_a = q
+        tmp_b = p
+
+        q = a_i_1 * q + b_i * prev_q
+        p = a_i_1 * p + b_i * prev_p
+        
+        prev_q = tmp_a
+        prev_p = tmp_b
+
+        # This is the hard part to compute.
+        # TODO - consider computing once in every X iters; Might be significant 
+        if i % divide_interval != 0:
+            continue
+        if q != 0:  # safety check
+            computed_values.append((mpmath.mpf(p) / mpmath.mpf(q), i))
+        else:
+            computed_values.append((mpmath.mpf(0), i))
+
+    global current_fig
```

Comment:
use convention g_current_fig for global variable

Reply:
updated & added comment